crates/typst/src/image.rs

```diff
@@ -306,55 +306,100 @@ fn decode_svg(
     Ok(Rc::new(DecodedImage::Svg(tree)))
 }
 
+/// A font family and its variants.
+#[derive(Clone)]
+struct FontData {
+    /// The usvg-compatible font family name.
+    usvg_family: EcoString,
+    /// The font variants included in the family.
+    fonts: EcoVec<Font>,
+}
+
 /// Discover and load the fonts referenced by an SVG.
 fn load_svg_fonts(
     tree: &usvg::Tree,
     loader: Tracked<dyn SvgFontLoader + '_>,
 ) -> fontdb::Database {
     let mut fontdb = fontdb::Database::new();
-    let mut referenced = BTreeMap::<EcoString, Option<EcoString>>::new();
+    let mut font_cache = BTreeMap::<EcoString, FontData>::new();
+    let mut loaded = BTreeMap::<EcoString, ()>::new();
```

Comment:
BTreeSet?

Reply:
Actually, in this case perhaps it just be `HashMap` and `HashSet`. Not sure whether I used BTreeMap here, we don't need any ordering guarantees or stable iteration order.

Reply:
Right, I don't know why I thought that sets didn't exist... I assume the BTreeMaps `WorldLoader::seen` and `PreparedLoader::families` can then also be replaced with HashMaps?

Reply:
Yeah, that's the problem. Because HashMap order is random, it can't be consistently hashed.

Reply:
This is the reason why BTreeMap is used quite often across Typst. But sometimes it's unnecessary after all.